### PR TITLE
add seperate multipath for worknodes.

### PR DESCRIPTION
**Pull request by npustchi**
*Issue number of the reported bug or feature request: #<number>*

**Describe your changes**
we need multipath enabled for worknodes and keep it disabled for other roles.

**Testing performed**
Created a BVE and changed the variabled to test the application on different type of nodes.

**Additional context**
created a jenkins run as well.

### CODE REVIEW

File: ansible/playbooks/roles/common/filter_plugins/util.py [hunk 1]
```diff
@@ -167,6 +182,7 @@ class FilterModule(object):
         'osadmin': osadmin,
         'etcdnode_to_cnames': etcdnode_to_cnames,
         'optimal_nvme_lbaf': optimal_nvme_lbaf,
+        'host_worknode': is_host_worknode,
```
[the remark this answers is not in the document]
Comment (npustchi) [reply ansible/playbooks/roles/common/filter_plugins/util.py:185]: I dont see a reason to call the jinja plug in to start with is, so decided to just call it, host_worknode.

File: ansible/playbooks/roles/common/defaults/main/base.yml [hunk 2]
```diff
@@ -313,7 +313,10 @@ hardware_io_scheduler: mq-deadline
 # Select kernel NVMe multipath support.
 # Disabling addresses the issue with controller vs. disk/device naming issues,
 # providing consistent, deterministic device naming conventions.
-hardware_nvme_multipath: false
+general_hardware_nvme_multipath: false
+
+# Compute node (worknode) specific NVMe multipath support
+compute_hardware_nvme_multipath: true
```
[the remark this answers is not in the document]
Comment (comay) [reply ansible/playbooks/roles/common/defaults/main/base.yml:319]: See the above - basically, I used the names you had but removed the `_hardware` substring.

File: ansible/playbooks/roles/common/filter_plugins/util.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (comay) [reply ansible/playbooks/roles/common/filter_plugins/util.py:185]: Hmm, okay - perhaps then the filter function itself should be renamed to `host_worknode` as well?

[the remark this answers is not in the document]
Comment (npustchi) [reply ansible/playbooks/roles/common/filter_plugins/util.py:185]: reason it is like this is, is_host_worknode follows python naming convention, but I could not find the same naming convention for filters in jinja.

File: ansible/playbooks/roles/common/defaults/main/base.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (npustchi) [reply ansible/playbooks/roles/common/defaults/main/base.yml:319]: yap makes sense, remove the hardware.

File: ansible/playbooks/roles/common/templates/grub/default.j2 [hunk 3]
```diff
@@ -19,7 +19,7 @@ GRUB_DISTRIBUTOR=`lsb_release -i -s 2> /dev/null || echo Debian`
 GRUB_CMDLINE_LINUX_DEFAULT="vga=normal nomodeset 3 console=tty0 console=tty1"
 {% set ignore_ce = 'mce=ignore_ce' if not hardware_edac_polling else '' -%}
 {% set ipv6_cmdline = 'ipv6.disable=1' if not system_enable_ipv6 else '' -%}
-{% set nvme_core_cmdline = 'nvme_core.multipath=' + ('1' if hardware_nvme_multipath else '0') -%}
+{% set nvme_core_cmdline = 'nvme_core.multipath=' + ('1' if (compute_hardware_nvme_multipath if hostvars[inventory_hostname] | host_worknode() else general_hardware_nvme_multipath) else '0') -%}
```
[the remark this answers is not in the document]
Comment (npustchi) [reply ansible/playbooks/roles/common/templates/grub/default.j2:22]: same as above.

**Review by comay — APPROVED** [review]

**Pull request merged**